app/models/mixins/class_from_string_mixin.rb

```diff
@@ -0,0 +1,12 @@
+module ClassFromStringMixin
+  extend ActiveSupport::Concern
+
+  module ClassMethods
+    def class_from_string(type)
+      return self if type == to_s || type.nil?
```

Comment:
I don't like the class_from_string method as it seems to imply you can get *any* class, but you really can't.  Perhaps, `descendant_by_name`  or `descendant_for`?  This seems to imply you will get the descendant class, and it mirrors the method named `descendants`